NSSLServer.Plugin.Recipes/Controller/RecipeController.cs

```diff
@@ -0,0 +1,309 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using NSSLServer.Database;
+using NSSLServer.Database.Attributes;
+using NSSLServer.Plugin.Shoppinglist.Manager;
+using System.Net.Http.Json;
+using System.Text.RegularExpressions;
+using static NSSLServer.Shared.RequestClasses;
+
+namespace NSSLServer.Plugin.Recipes.Controller
+{
+    [Route("recipe"), WithDbContext]
+    public class RecipeController : AuthenticatingDbContextController
+    {
+        private const string BaseUrl = "https://api.chefkoch.de/v2/recipes/";
+
+        private static readonly Regex IdRegex = new(@"(\d{10,})", RegexOptions.IgnoreCase);
+
+        private readonly HttpClient _httpClient;
+
+        public RecipeController(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
+
+        [HttpPost, Route(nameof(CreateShoppingListForRecipe))]
+        public async Task<IActionResult> CreateShoppingListForRecipe([FromBody] AddRecipeArgs urlOrId, string? optionalListName = null, int? amountOfPeople = 4)
```

Comment:
Another cool idea would be "AddRecipeToList" Method, which would be nearly the same as this method with the exception, that no new list is added. The parameter optionalListName would be replaced with a mandatory listid